apacheconfig/wloader.py

```diff
@@ -0,0 +1,160 @@
+#
+# This file is part of apacheconfig software.
+#
+# Copyright (c) 2018-2019, Ilya Etingof <etingof@gmail.com>
+# License: https://github.com/etingof/apacheconfig/LICENSE.rst
+#
+
+
+from apacheconfig import make_parser
+from apacheconfig import make_lexer
+
+import abc
+import six
+
+
+def _restore_original(word):
+    """If the `word` is a Quoted string, restores it to original.
+    """
+    if getattr(word, 'is_single_quoted', False):
+        return "'%s'" % word
+    if getattr(word, 'is_double_quoted', False):
+        return '"%s"' % word
+    return word
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Node():
+    """Generic class containing data that represents a node in the config AST.
+    """
+
+    @abc.abstractmethod
+    def __str__(self):
+        raise NotImplementedError()
+
+    @property
+    def type(self):
+        """A typestring as defined by the apacheconfig parser.
+
+        A single *Node class can have multiple possible values for this. For
```

Comment:
Should we have these values enumerated somewhere?

Reply:
I just did this in the documentation for the time being, but in the future maybe we can do this with #68 as well.